pkg/saga/statemachine/engine/invoker/grpc_invoker_test.go

```diff
@@ -0,0 +1,168 @@
+package invoker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/seata/seata-go/pkg/saga/statemachine/statelang/state"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"testing"
+	"time"
+
+	pb "github.com/seata/seata-go/testdata/saga/engine/invoker/grpc"
+)
+
+type MockGPRCClientImpl struct {
+	GPRCClientImpl
+}
+
+type mockClientImpl struct {
+	invokeCount int
+}
+
+func (m *mockClientImpl) SayHelloRight(ctx context.Context, word string) (string, error) {
+	m.invokeCount++
+	fmt.Println("invoke right")
+	return word, nil
+}
+
+func (m *mockClientImpl) SayHelloRightLater(ctx context.Context, word string, delay int) (string, error) {
+	m.invokeCount++
+	if delay == m.invokeCount {
+		fmt.Println("invoke right")
+		return word, nil
+	}
+	fmt.Println("invoke fail")
+	return "", errors.New("invoke failed")
+}
+
+func TestGRPCInvokerInvokeSucceedWithOutRetry(t *testing.T) {
+	ctx := context.Background()
+	invoker := newGRPCServiceInvoker()
+	values, err := invoker.Invoke(ctx, []any{"hello"}, newHelloServiceTaskState())
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	if values == nil || len(values) == 0 {
+		t.Error("no value in values")
+		return
+	}
+	if values[0].Interface().(string) != "hello" {
+		t.Errorf("expect hello, but got %v", values[0].Interface())
+	}
+	if _, ok := values[1].Interface().(error); ok {
+		t.Errorf("expect nil, but got %v", values[1].Interface())
+	}
+}
+
+func TestGRPCInvokerInvokeSucceedInRetry(t *testing.T) {
+	ctx := context.Background()
+	invoker := newGRPCServiceInvoker()
+	values, err := invoker.Invoke(ctx, []any{"hello", 2}, newHelloServiceTaskStateWithRetry())
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	if values == nil || len(values) == 0 {
+		t.Error("no value in values")
+		return
+	}
+	if values[0].Interface().(string) != "hello" {
+		t.Errorf("expect hello, but got %v", values[0].Interface())
+	}
+	if _, ok := values[1].Interface().(error); ok {
+		t.Errorf("expect nil, but got %v", values[1].Interface())
+	}
+}
+
+func TestGRPCInvokerInvokeFailedInRetry(t *testing.T) {
+	ctx := context.Background()
+	invoker := newGRPCServiceInvoker()
+	_, err := invoker.Invoke(ctx, []any{"hello", 5}, newHelloServiceTaskStateWithRetry())
+	if err != nil {
+		assert.Error(t, err)
+	}
+}
+
+func TestGRPCInvokerInvokeE2E(t *testing.T) {
+	go func() {
+		pb.StartProductServer()
+	}()
+	time.Sleep(3000)
```

Comment:
The default unit is nanoseconds. Is 3000 ns (aka. 3 us) an expected value?

Reply:
Ok, I have set it to ms.